src/models.js

```diff
@@ -70,12 +70,17 @@ const elemFaces = {
 }
 
 function renderLiquid (world, cursor, texture, type, water, attr) {
+  const blockAbove = world.getBlock(cursor.plus(new Vec3(0, 1, 0)))
+  const liquidHeight = (blockAbove.type === type ? 16 : 14) / 16
+
   for (const face in elemFaces) {
     const { dir, corners } = elemFaces[face]
 
     const neighbor = world.getBlock(cursor.plus(dir))
```

Comment:
dir is not a vec3 anymore with your latest change, so plus() will fail

Reply:
Fixed in 241895917ee18d646f16aa77ee224a8f425d9bc1.